wavplayer.cpp

```diff
@@ -416,6 +416,40 @@ std::vector<float> highpassFilterAudio(
     return modifiedData;
 }
 
+// get audio data to vector<float>
+std::vector<float> extractAudioData(
```

Comment:
I added a new function to just get the audio data into a vector<float> when downloading (if pitch-shift isn't used).  We do this here but not in the playback function because playback uses the audiospec struct which holds the audio data already on wav load. Without it, there wouldn't be any audio data to pack into a wav file. 

I was originally thinking this function could also be used in the playback function (e.g. get all the audio data first as float, then pass the data through the filters) but I had trouble getting it working with the pitch-shift function. I was getting some seg faults sometimes and I wonder if it had to do with the cvt.buf data - e.g. the pitch shift function maybe needs to work directly with the cvt.buf bytes rather than if I tried to push all those bytes into a vector<float>?